src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -91,231 +66,68 @@ const getSafeTxGas = async (txProps: RequiredTxProps, safeVersion: string): Prom
   return safeTxGas
 }
 
-export class TxSender {
-  notifications: ReturnType<typeof createTxNotifications>
-  nonce: string
-  isFinalization: boolean
-  txArgs: TxArgs
-  safeTxHash: string
-  txProps: RequiredTxProps
-  from: string
-  dispatch: Dispatch
-  safeInstance: GnosisSafe
-  safeVersion: string
-  txId: string
-
-  // On transaction completion (either confirming or executing)
-  async onComplete(signature?: string, confirmCallback?: ConfirmEventHandler): Promise<void> {
-    const { txArgs, safeTxHash, txProps, dispatch, notifications, isFinalization } = this
-
-    // Propose the tx to the backend
-    // 1) If signing
-    // 2) If creating a new tx (no txId yet)
-    let txDetails: TransactionDetails | null = null
-    if (!isFinalization || !this.txId) {
-      try {
-        txDetails = await saveTxToHistory({ ...txArgs, signature, origin })
-      } catch (err) {
-        logError(Errors._816, err.message)
-        return
-      }
-    }
-
-    // If threshold reached except for last sig, and owner chooses to execute the created tx immediately
-    // we retrieve txId of newly created tx from the proposal response
-    if (isFinalization && txDetails) {
-      dispatch(addPendingTransaction({ id: txDetails.txId }))
-    }
-
-    notifications.closePending()
-
-    // This is used to communicate the safeTxHash to a Safe App caller
-    confirmCallback?.(safeTxHash)
-
-    // Go to a tx deep-link
-    if (txDetails && txProps.navigateToTransactionsTab) {
-      navigateToTx(txProps.safeAddress, txDetails)
-    }
-
-    dispatch(fetchTransactions(_getChainId(), txProps.safeAddress))
-  }
-
-  async onError(err: Error & { code: number }, errorCallback?: ErrorEventHandler): Promise<void> {
-    const { txArgs, isFinalization, from, txProps, dispatch, notifications, safeInstance, txId } = this
-
-    logError(Errors._803, err.message)
-
-    errorCallback?.()
-
-    notifications.closePending()
-
-    // Existing transaction was being finalised (txId exists)
-    if (isFinalization && txId) {
-      dispatch(removePendingTransaction({ id: txId }))
-    }
-
-    const executeDataUsedSignatures = safeInstance.methods
-      .execTransaction(
-        txProps.to,
-        txProps.valueInWei,
-        txProps.txData,
-        txProps.operation,
-        0,
-        0,
-        0,
-        ZERO_ADDRESS,
-        ZERO_ADDRESS,
-        txArgs.sigs,
-      )
-      .encodeABI()
-    const contractErrorMessage = await fetchOnchainError(executeDataUsedSignatures, safeInstance, from)
-
-    notifications.showOnError(err, contractErrorMessage)
+const getCreationTxArgs = async (
```

Comment:
Nit: isn't it more correct to say "tx creation" instead of "creation tx"? to me, creation tx sounds more like a transaction to create something, but here we create the transaction itself